sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsClientBuilder.java

```diff
@@ -0,0 +1,345 @@
+ * {@link #buildAsyncClient() buildAsyncClient} respectively to construct an instance of the desired client.
+ *
+ * <p>The client needs the service endpoint of the Azure Text Analytics to access the resource service.
+ * {@link #subscriptionKey(String) subscriptionKey(String)} or
+ * {@link #credential(TokenCredential) credential(TokenCredential)} give the builder access credential. </p>
+ *
+ * <p>Another way to construct the client is using a {@link HttpPipeline}. The pipeline gives the client an
+ * authenticated way to communicate with the service. Set the pipeline with {@link #pipeline(HttpPipeline) this} and
+ * set the service endpoint with {@link #endpoint(String) this}. Using a
+ * pipeline requires additional setup but allows for finer control on how the {@link TextAnalyticsClient} and
+ * {@link TextAnalyticsAsyncClient} is built.</p>
+ *
+ * @see TextAnalyticsAsyncClient
+ * @see TextAnalyticsClient
+ */
+@ServiceClientBuilder(serviceClients = {TextAnalyticsAsyncClient.class, TextAnalyticsClient.class})
+public final class TextAnalyticsClientBuilder {
+    private static final String ECHO_REQUEST_ID_HEADER = "x-ms-return-client-request-id";
+    private static final String CONTENT_TYPE_HEADER = "Content-Type";
+    private static final String CONTENT_TYPE_HEADER_VALUE = "application/json";
+    private static final String ACCEPT_HEADER = "Accept";
+    private static final String ACCEPT_HEADER_VALUE = "application/json";
+    private static final String TEXT_ANALYTICS_PROPERTIES = "azure-textanalytics.properties";
+    private static final String OCP_APIM_SUBSCRIPTION_KEY = "Ocp-Apim-Subscription-Key";
+    private static final String NAME = "name";
+    private static final String VERSION = "version";
+    private static final RetryPolicy DEFAULT_RETRY_POLICY = new RetryPolicy("retry-after-ms", ChronoUnit.MILLIS);
+    private static final String DEFAULT_SCOPE = "https://cognitiveservices.azure.com/.default";
+
+    private final ClientLogger logger = new ClientLogger(TextAnalyticsClientBuilder.class);
+    private final List<HttpPipelinePolicy> policies;
+    private final HttpHeaders headers;
+    private final String clientName;
+    private final String clientVersion;
+
+    private String endpoint;
+    private String subscriptionKey;
+    private TokenCredential tokenCredential;
+    private HttpClient httpClient;
+    private HttpLogOptions httpLogOptions;
+    private HttpPipeline httpPipeline;
+    private Configuration configuration;
+    private HttpPipelinePolicy retryPolicy;
+    private TextAnalyticsClientOptions clientOptions;
+    private TextAnalyticsServiceVersion version;
+
+    /**
+     * The constructor with defaults.
+     */
+    public TextAnalyticsClientBuilder() {
+        policies = new ArrayList<>();
+        httpLogOptions = new HttpLogOptions();
+
+        Map<String, String> properties = CoreUtils.getProperties(TEXT_ANALYTICS_PROPERTIES);
```

Comment:
I noticed in the recording for tests, the SDK version and name were both UnknownName/UnknownVersion. Is this correct?

Reply:
Updated  to 
 ```private static final String TEXT_ANALYTICS_PROPERTIES = "azure-ai-textanalytics.properties";```
to both here and TextAnalyticsTestBase  should solve this error